experimental/PlaneCurve/ProjCurve.jl

```diff
@@ -21,6 +50,222 @@ end
+
+@doc Markdown.doc"""
+    reduction(C::ProjCurve)
+
+Return the projective curve defined by the radical of the defining ideal of `C`.
+
+# Example
+```jldoctest
+julia> S, (x, y, z, t) = PolynomialRing(QQ, ["x", "y", "z", "t"])
+(Multivariate Polynomial Ring in x, y, z, t over Rational Field, fmpq_mpoly[x, y, z, t])
+
+julia> T, _ = grade(S)
+(Multivariate Polynomial Ring in x, y, z, t over Rational Field graded by
+  x -> [1]
+  y -> [1]
+  z -> [1]
+  t -> [1], MPolyElem_dec{fmpq, fmpq_mpoly}[x, y, z, t])
+
+julia> I = ideal(T, [x^2, y^2*z, z^2])
+ideal(x^2, y^2*z, z^2)
+
+julia> C = Oscar.ProjCurve(I)
+Projective curve defined by the ideal(x^2, y^2*z, z^2)
+
+julia> Oscar.reduction(C)
+Projective curve defined by the ideal(z, x)
+```
+"""
+function reduction(C::ProjCurve)
+    J = radical(Oscar.defining_ideal(C))
+    return ProjCurve(J)
+end
+
+################################################################################
+
+@doc Markdown.doc"""
+    jacobi_ideal(C::ProjCurve)
+
+Return the jacobian ideal of the defining ideal of `C`.
+
+# Example
+```jldoctest
+julia> S, (x, y, z, t) = PolynomialRing(QQ, ["x", "y", "z", "t"])
+(Multivariate Polynomial Ring in x, y, z, t over Rational Field, fmpq_mpoly[x, y, z, t])
+
+julia> T, _ = grade(S)
+(Multivariate Polynomial Ring in x, y, z, t over Rational Field graded by
+  x -> [1]
+  y -> [1]
+  z -> [1]
+  t -> [1], MPolyElem_dec{fmpq, fmpq_mpoly}[x, y, z, t])
+
+julia> I = ideal(T, [x^2, y^2*z, z^2])
+ideal(x^2, y^2*z, z^2)
+
+julia> C = Oscar.ProjCurve(I)
+Projective curve defined by the ideal(x^2, y^2*z, z^2)
+
+julia> Oscar.jacobi_ideal(C)
+ideal(4*x*y*z, 2*x*y^2, 4*x*z, 4*y*z^2)
+```
+"""
+function Oscar.jacobi_ideal(C::ProjCurve)
+    I = Oscar.defining_ideal(C)
+    R = base_ring(I)
+    k = C.ambiant_dim - 1
+    M = jacobi_matrix(I.gens.O)
+    V = minors(M, k)
+    filter!(!iszero, V)
+    return ideal(R, V)
+end
+
+################################################################################
+
+@doc Markdown.doc"""
+    invert_birational_map(phi::Vector{T}, C::ProjCurve) where {T <: MPolyElem}
+
+Return a dictionary where `image` represents the image of the birational map
+given by `phi`, and `inverse` represents its inverse, where `phi` is a
+birational map of the projective curve `C` to its image in the projective
+space of dimension `size(phi) - 1`.
+Note that the entries of `inverse` should be considered as
+representatives of elements in `R/image`, where `R` is the basering.
+"""
+function invert_birational_map(phi::Vector{T}, C::ProjCurve) where {T <: MPolyElem}
+    s = parent(phi[1])
+    I = ideal(s, phi)
+    singular_assure(I)
+    IC = defining_ideal(C)
+    singular_assure(IC)
+    L = Singular.LibParaplanecurves.invertBirMap(I.gens.S, IC.gens.S)
+    R = _fromsingular_ring(L[1])
+    J = L[2][:J]
+    psi = L[2][:psi]
+    return Dict([("image", gens(ideal(R, J))), ("inverse", gens(ideal(R, psi)))])
```

Comment:
This would be more idiomatic:
```suggestion
    return Dict(:image => gens(ideal(R, J), "inverse => gens(ideal(R, psi)))
```